### ISSUE TITLE
Run the website locally

### ISSUE THREAD

**Issue by micheleminno** [humanetech-community/community-hub#14]
Right now only @aschrijver has access/write privileges to upload changes on the server.
But anyone can run the site locally and see how the changes done on the code apply
(Full instructions [here](https://jekyllrb.com/)). 

Once you created the local folder for the website, you have to clone the code here in the community-hub repository:

`git clone https://github.com/humanetech-community/community-hub.git`

Install the missing libraries:

`bundle install`

 and put the website up:

`bundle exec jekyll serve`

To see the website go to:

http://localhost:4000

**Comment by micheleminno** [comment]
Is that right, @aschrijver ?

**Comment by aschrijver** [comment]
> Right now only @aschrijver has access/write privileges to upload changes on the server.

This is not accurate. There are 3 maintainers of this repository, @micheleminno @gkrishnaks and @aschrijver which means we can make direct commits to the repo.

But anyone else can fork the repository, commit changes to their fork, then create a Pull Request (PR) to have the changes be reviewed by one of the maintainers who can merge the changes to the production website.

**Comment by aschrijver** [comment]
If you install Jekyll, then you can start local development with this command:

`jekyll serve --incremental`

If you make changes to the Navigation or Stylesheets (SaSS) then you need to stop Jekyll, and do a

`jekyll clean`

before starting again, to ensure you see your changes.

This website is based on the following project, where you will find a lot of instructions on how to make changes: https://mmistakes.github.io/minimal-mistakes/

**Comment by micheleminno** [comment]
> Right now only @aschrijver has access/write privileges to upload changes on the server.
> 
> This is not accurate. There are 3 maintainers of this repository, @micheleminno @gkrishnaks and @aschrijver which means we can make direct commits to the repo.

I don't mean the github repo, I mean the actual server where https://humanetech.community runs from (is there such a server? Or is it run directly from the github repo?)

**Comment by micheleminno** [comment]
By the way I managed to run the website locally, I'm working on #15

**Comment by aschrijver** [comment]
> I don't mean the github repo, I mean the actual server where https://humanetech.community runs from (is there such a server? Or is it run directly from the github repo?)

Github is running this server. This is part of the Github Pages feature. That is why this option - using Jekyll - is so attractive. So I do not have a server that runs this.

**Comment by micheleminno** [comment]
Ok, so every time anyone of the three of us commits something, it automatically updates the website. As it does locally if you installed Jekyll locally.

**Comment by aschrijver** [comment]
Yes indeed. And the same occurs when accepting + merging a PR of someone else.

**Comment by micheleminno** [comment]
Ok, just to clarify all contributing process. I can close this.

**Comment by aschrijver** [comment]
We can document procedures in the README.md so it is clear for others. Could make that a separate task issue.

**Comment by aschrijver** [comment]
> Once you created the local folder for the website, you have to clone the code here in the community-hub repository:
>
> git clone https://github.com/humanetech-community/community-hub.git

Note: This only works if you are a maintainer of the repository. Otherwise you should first create a Fork of the repository, and then clone that fork to a local folder. You then have the option to commit changes to your fork and from the Github UI create a Pull Request with these changes to the original repo.

This procedure needs to be elaborated a bit further, especially the part to synchronize your fork to changes in the original (the easiest way to do this, is to delete your fork, and then fork a new version).
